descartes_light/include/descartes_light/impl/descartes_light.hpp

```diff
@@ -111,7 +111,8 @@ bool Solver<FloatType>::build(const std::vector<typename PositionSampler<FloatTy
     const auto& from = graph_.getRung(static_cast<size_t>(i) - static_cast<size_t>(1));
     const auto& to = graph_.getRung(static_cast<size_t>(i));
 
-    if (!edge_eval->evaluate(from, to, graph_.getEdges(static_cast<size_t>(i) - static_cast<size_t>(1))))
+    if (!edge_eval[static_cast<size_t>(i - 1)]->evaluate(
```

Comment:
Unrelated, but why is i a long if we always cast it to a size_t?

Reply:
Currently size_t is not supported by omp macros so it must be of type long or int.